### PR TITLE
Fix chat/topic validation

### LINKED ISSUE #5117: No validation message if you are trying to add invalid contact code or Join public chat with invalid name
### User Story

As a user, I want to see what I did wrong if I entered invalid chat name or contact code

### Description

*Type*: Feature

*Summary*: I want to see button enabled (if I insert invalid contact code in "New Chat" or set the wrong topic in "Join Public Chat"( i.e., #1234)) and after clicking - corresponding validation error, like in mobile client.

#### Expected behavior
*Design:* zpl.io/aB5mnom

#### Actual behavior
"Start chat" and "Public Chat" are not enabled when values don't fit validation

#### Reproduction

1. Open StatusIm, log in
2. Click on +
3. Enter invalid contact key (i.e. '123') in 'Add new contact', click on 'Add'
4. Repeat 2,  click on 'Add' in 'Join to public chat' |(with empty value)
5. Repeat 2, enter an invalid topic name (i. e. '123') in 'Join to public chat', click on 'Add'

### Additional Information
MacBook Pro (Retina, 15-inch, Mid 2015)
*OS:* Mac OSx High Sierra 10.13.5
*Build:* Status version: desktop build 10/07/18 (after merge https://github.com/status-im/status-react/pull/4752)

### Solution
Design and cases here >> zpl.io/bAEEv3B (Zeplin, needs permissions)
Example:
captura de pantalla 2018-08-20 a la s 8 52 44 p m

### LINKED ISSUE #5849: Can start chat with non-existing user
### Description

*Type*: Bug
*Summary*: "Start chat" button is always enabled despite of content of "Contact code" field

#### Expected behavior
"Start chat" button is not available until you enter valid contact key value  or if you enter wrong value - validation error is shown (as it mentioned in https://github.com/status-im/status-react/issues/5117)

#### Actual behavior
can start chat with non-existing user
act

### Reproduction
[comment]: # (Describe how we can replicate the bug step by step.)

- Open Status
- Tap on +
- Enter to contact code field invalid value (i. e. "jjjjj")
- Tap on Start chat

### Additional Information

* Status version: nightly 14/09/18
* Operating System: Mac OSx
* Video: http://take.ms/nMFCck

<blockquote><div>Monosnap screenshot tool</div><div><strong><a href="https://monosnap.com/file/EJG3hycTiqfZSTBIH6o0qfsQ3zHKNQ">File "screencast 2018-09-14 16-16-35.mp4"</a></strong></div><div>Monosnap — the best tool to take & share your screenshots.</div></blockquote>

### PR DESCRIPTION
**Pull request by vitvly**
Fixes #5117
Fixes #5849

### CODE REVIEW

**Comment by ghost** [comment]
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status.html))?

File: src/status_im/ui/screens/desktop/main/add_new/views.cljs [hunk 1]
```diff
@@ -67,29 +80,32 @@
...
-                                                     [_ before after] (first (re-seq #"(.*)\#(.*)" text))]
-                                                 (.setNativeProps @topic-input-ref (js-obj "text" (str "#" before after)))
-                                                 (re-frame/dispatch [:set :public-group-topic (subs text 1)])))}]]
-       [react/touchable-highlight {:disabled topic-error
-                                   :on-press #(when-not topic-error
-                                                (do
+      (let [disable? (or (not (string/blank? topic-error))
+                         (string/blank? topic))]
+        [react/view {:style styles/add-contact-edit-view}
+         [react/view {:flex 1}
```
Comment (rasom) [inline src/status_im/ui/screens/desktop/main/add_new/views.cljs]: Are you sure it actually has any effect? I found that some style props will be applied only if passed in `styles`.

**Review by vkjr — APPROVED** [review]

File: src/status_im/ui/screens/desktop/main/add_new/views.cljs (hunk 1 above)
Comment (vitvly) [reply src/status_im/ui/screens/desktop/main/add_new/views.cljs]: Yes, I specifically checked that `{:flex 1}` does what I need. Will have to dig to the root cause of this `{:style ...}` problem at some point.

**Review by maxhora — APPROVED** [review]

**Comment by churik** [comment]
Tested public chat  name an contact code fields validation for IOS, Android, Mac OSx.

**Pull request merged**